### ISSUE TITLE
jupyterhub redirect loop in k8s

### ISSUE THREAD

**Issue by stczwd** [jupyterhub/jupyterhub#1783]
I'm using jupyterhub in k8x, using default SudoSpawner.
I have packaged all nessary environment to a single directory, just unpackaging notebook.tar and running start.sh can start a new jupyterhub service.I've already tried it on the server and docker, and it worked. But in k8x, it never work, just redirect between `/user/admin` and `/hub/user/admin`.
This is the log:
```
[I 2018-04-08 11:28:00.736 JupyterHub base:751] status:None
11:28:00.834 - debug: [ConfigProxy] PROXY WEB /user/admin/?redirects=1 to http://172.30.4.8:8081
[I 2018-04-08 11:28:00.836 JupyterHub log:122] 302 GET /hub/user/admin/ ¡ú /notebook/user/admin/?redirects=1 (admin@172.30.17.1) 2607.72ms
[I 2018-04-08 11:28:00.838 JupyterHub log:122] 302 GET /user/admin/?redirects=1 ¡ú /hub/user/admin/?redirects=1 (@172.30.17.1) 0.96ms
11:28:00.873 - debug: [ConfigProxy] PROXY WEB /hub/user/admin/?redirects=1 to http://172.30.4.8:8081
[I 2018-04-08 11:28:00.938 JupyterHub base:723] spawner.pending:False, spawner._spawn_future:None
[I 2018-04-08 11:28:00.939 JupyterHub base:734] spawner.pending:False
[I 2018-04-08 11:28:00.939 JupyterHub base:744] spawner.pending:True
[I 2018-04-08 11:28:00.939 JupyterHub base:751] status:None
[W 2018-04-08 11:28:01.035 JupyterHub base:805] Redirect loop detected on /hub/user/admin/?redirects=1
[I 2018-04-08 11:28:03.042 JupyterHub log:122] 302 GET /hub/user/admin/?redirects=1 ¡ú /user/admin/?redirects=2 (admin@172.30.17.1) 2104.61ms
11:28:03.148 - debug: [ConfigProxy] PROXY WEB /user/admin/?redirects=2 to http://172.30.4.8:8081
[I 2018-04-08 11:28:03.152 JupyterHub log:122] 302 GET /user/admin/?redirects=2 ¡ú /hub/user/admin/?redirects=2 (@172.30.17.1) 1.13ms
11:28:03.183 - debug: [ConfigProxy] PROXY WEB /hub/user/admin/?redirects=2 to http://172.30.4.8:8081
[I 2018-04-08 11:28:03.234 JupyterHub base:723] spawner.pending:False, spawner._spawn_future:None
[I 2018-04-08 11:28:03.234 JupyterHub base:734] spawner.pending:False
[I 2018-04-08 11:28:03.234 JupyterHub base:744] spawner.pending:True
[I 2018-04-08 11:28:03.235 JupyterHub base:751] status:None
[W 2018-04-08 11:28:03.239 JupyterHub base:805] Redirect loop detected on /hub/user/admin/?redirects=2
```
In `jupyterhub_configure.py`, I changed hub_ip, proxy_ip and spawner_ip to docker container's ip.
I have seen the details:
1. singleuser subprocess is processing. I can use subprocess's ip and port curl this service
2. when redirect /user/admin, the program allways goto PrefixRedirectHandler, instead of AddSlashHandler or RedirectWithParams.
3. user's token or cookie is correct
4. the route to subprocess has been created correctly

Anybody can tell me how this happens?

**Comment by willingc** [comment]
@minrk is this a Unicode error in the log that we fixed back in January (though not in 0.8.1)? Or related to the User Spawner handlers?

**Comment by minrk** [comment]
@stczwd what version of configurable-http-poryx do you have? Can you show more complete logs, including when the server is started? If the server is running, `/user/admin/` shouldn't be sent to the Hub, which means that either the proxy has the wrong information or there is a bug in the proxy.

There [was a bug](https://github.com/jupyterhub/configurable-http-proxy/pull/123) in CHP which would route URLs that look similar to this incorrectly, which is fixed in CHP 3.0

**Comment by stczwd** [comment]
@minrk @willingc Thanks for your attention.I've found the reason, but it's not in the Jupyterhub.
In my case, there's an openresty used to find notebook service. It redirects the /notebook request, which is the jupyterhub's base_url,  to http://xxx/notebook/. 
That means when user request /notebook/user/xxx, it redirects to  /notebook//user/xxx, which makes configurable-http-configurable failed to transfer the link to the right handler.
